packages/python-packages/api-stub-generator/apistub/nodes/_pylint_parser.py

```diff
@@ -0,0 +1,81 @@
+import inspect
+import json
+import os
+from pylint import epylint
+import re
+from typing import List
+
+_HELP_LINK_REGEX = re.compile(r"(.+) See details: *([^\s]+)")
+
+class PylintError:
+
+    def __init__(self, pkg_name, **kwargs):
+        from apistub import DiagnosticLevel
+        self.type = kwargs.pop('type', None)
+        self.module = kwargs.pop('module', None)
+        self.obj = kwargs.pop('obj', None)
+        self.line = kwargs.pop('line', None)
+        self.column = kwargs.pop('column', None)
+        self.end_line = kwargs.pop('endLine', None)
+        self.end_column = kwargs.pop('endColumn', None)
+        self.path = kwargs.pop('path', None)
+        self.symbol = kwargs.pop('symbol', None)
+        self.message = kwargs.pop('message', None)
+        self.message_id = kwargs.pop('message-id', None)
+        self.help_link = None
+        if self.path.startswith(pkg_name):
+            self.path = self.path[(len(f"{pkg_name}\\\\") - 1):]
+        code = self.symbol[0]
+        self.level = DiagnosticLevel.ERROR if code in "EF" else DiagnosticLevel.WARNING
+        self._parse_help_link()
+
+    def _parse_help_link(self):
+        try:
+            (message, help_link) = _HELP_LINK_REGEX.findall(self.message)[0]
+            self.message = message
+            self.help_link = help_link
+        except Exception as err:
+            # if unable to parse, leave alone
+            return
+        
+
+    def generate_tokens(self, apiview, target_id):
+        apiview.add_diagnostic(obj=self, target_id=target_id)
+
+
+class PylintParser:
+
+    AZURE_CHECKER_CODE = "47"
+
+    items: List[PylintError] = []
+
+    @classmethod
+    def parse(cls, path):
+        from apistub import ApiView
+
+        pkg_name = os.path.split(path)[-1]
+        rcfile_path = os.path.join(ApiView.ROOT, "azure_sdk_pylintrc")
+        (pylint_stdout, pylint_stderr) = epylint.py_run(f"{path} -f json --rcfile {rcfile_path}", return_std=True)
```

Comment:
This is cool - I didn't know pylint could do this :)

Reply:
Well, it's not _that_ cool, because their JSON stdout can be interspersed with Exceptions that render it non-JSON 😆 but nothing we can't work around.